x/gamm/pool-models/balancer/pool_test.go

```diff
@@ -1217,3 +1262,114 @@ func TestIsActive(t *testing.T) {
+	balancerPoolAssets := []balancer.PoolAsset{
+		{Token: sdk.NewInt64Coin("foo", 100), Weight: sdk.NewIntFromUint64(5)},
+		{Token: sdk.NewInt64Coin("bar", 100), Weight: sdk.NewIntFromUint64(5)},
+	}
+
+	tests := map[string]struct {
+		tokensIn        sdk.Coins
+		poolAssets      []balancer.PoolAsset
+		expNumShare     sdk.Int
+		expTokensJoined sdk.Coins
+		expPoolAssets   []balancer.PoolAsset
+		expectPass      bool
+	}{
+		"two asset pool, same tokenIn ratio": {
+			tokensIn:        sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(10)), sdk.NewCoin("bar", sdk.NewInt(10))),
+			poolAssets:      balancerPoolAssets,
+			expNumShare:     sdk.NewIntFromUint64(10000000000000000000),
+			expTokensJoined: sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(10)), sdk.NewCoin("bar", sdk.NewInt(10))),
+			expPoolAssets:   balancerPoolAssets,
+			expectPass:      true,
+		},
+		"two asset pool, different tokenIn ratio with pool": {
+			tokensIn:        sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(10)), sdk.NewCoin("bar", sdk.NewInt(11))),
+			poolAssets:      balancerPoolAssets,
+			expNumShare:     sdk.NewIntFromUint64(10000000000000000000),
+			expTokensJoined: sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(10)), sdk.NewCoin("bar", sdk.NewInt(10))),
+			expPoolAssets:   balancerPoolAssets,
+			expectPass:      true,
+		},
+		"two asset pool, no-swap join attempt with one asset": {
+			tokensIn:        sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(10))),
+			poolAssets:      balancerPoolAssets,
+			expNumShare:     sdk.NewIntFromUint64(0),
+			expTokensJoined: sdk.Coins{},
+			expPoolAssets:   balancerPoolAssets,
+			expectPass:      false,
+		},
+		"two asset pool, no-swap join attempt with one valid and one invalid asset": {
+			tokensIn:        sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(10)), sdk.NewCoin("baz", sdk.NewInt(10))),
+			poolAssets:      balancerPoolAssets,
+			expNumShare:     sdk.NewIntFromUint64(0),
+			expTokensJoined: sdk.Coins{},
+			expPoolAssets:   balancerPoolAssets,
+			expectPass:      false,
+		},
+		"two asset pool, no-swap join attempt with two invalid assets": {
+			tokensIn:        sdk.NewCoins(sdk.NewCoin("baz", sdk.NewInt(10)), sdk.NewCoin("qux", sdk.NewInt(10))),
+			poolAssets:      balancerPoolAssets,
+			expNumShare:     sdk.NewIntFromUint64(0),
+			expTokensJoined: sdk.Coins{},
+			expPoolAssets:   balancerPoolAssets,
+			expectPass:      false,
+		},
+		"single asset pool, no-swap join attempt with one asset": {
+			tokensIn: sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(sdk.MaxSortableDec.TruncateInt64()))),
+			poolAssets: []balancer.PoolAsset{
+				{Token: sdk.NewCoin("foo", sdk.NewInt(1)), Weight: sdk.NewIntFromUint64(1)},
+			},
+			expNumShare:     sdk.NewIntFromUint64(0),
+			expTokensJoined: sdk.Coins{},
+			expPoolAssets: []balancer.PoolAsset{
+				{Token: sdk.NewCoin("foo", sdk.NewInt(1)), Weight: sdk.NewIntFromUint64(1)},
+			},
+			expectPass: false,
+		},
+		"duplicate asset pool, no-swap join attempt with duplicate assets": {
```

Comment:
Does it make sense to have a test case for 3 assets?

Reply:
Added three-asset versions of the core test cases!